qmi/core/transport.py

```diff
@@ -647,18 +630,38 @@ def read(self, nbytes: int, timeout: Optional[float]) -> bytes:
         while nbuf < nbytes:
             self._safe_socket.settimeout(tremain)
             try:
-                b = self._safe_socket.recv(nbytes - nbuf)
+                if self._assert_addr:
+                    # For UDP, reading data fails if the arriving packet size is larger than the given nbytes (- nbuf)
+                    # to read. To avoid this happening, read at least 4096 standard that is larger than regular MTU.
+                    udp_packet_size = max(nbytes - nbuf, 4096)
```

Comment:
In those cases the client should have some data fragmenting set-up, which divides the large packets into max 4096 bytes. Otherwise, the UDP transport cannot be used for those cases. TCP would work.

Reply:
BTW I noticed different behaviour between unit-tests on my laptop and on the repo runner. Seems like the repo server fragments the packets while locally it does not happen. Could be also a Windows/Linux difference. But the unit-tests should be fine now and work for both.